lib/orogen/templates/main.cpp

```diff
@@ -439,6 +439,35 @@ RTT::internal::GlobalEngine::Instance(ORO_SCHED_OTHER, RTT::os::LowestPriority);
     <% end %>
 
     exiting = false;
+
+    if (vm.count("ior-write-fd")) {
+        int ior_write = vm["ior-write-fd"].as<int>();
+        std::string ior;
+        std::ostringstream message_ostream;
+        int write_result = 0;
+
+        message_ostream << "{";
+    <% activity_ordered_tasks.each do |task| %>
+        ior = RTT::corba::TaskContextServer::getIOR(task_<%= task.name %>.get());
+        message_ostream << "\"" << task_<%= task.name %>.get()->getName() << "\": \"" << ior << "\"";
+    <% if task != activity_ordered_tasks.last %>
+        message_ostream << ",";
+    <% end %>
+    <% end %>
+        message_ostream << "}";
+        std::string message = message_ostream.str();
+        while (write_result < message.length() + 1) {
+            message.erase(0, write_result);
+            write_result = write(ior_write, message.c_str(), message.length() + 1);
```

Comment:
You don't need the +1 (actually, the +1 is harmful). You definitely don't want to send the NULL byte.